// The Firmament project
// Copyright (c) 2013 Malte Schwarzkopf <malte.schwarzkopf@cl.cam.ac.uk>
//
// Protobuf representing the capacities, weights and costs for the flow-based
// scheduler.

package firmament;

message SchedulingParameters {
  optional FlowSchedulingFactors factors = 1;
//  optional FlowScheduli
}

message FlowSchedulingFactors {
  optional double wait = 1 [default = 0.5];             // Quincy: \omega
  optional double core_switch = 2 [default = 2.0];      // Quincy: \xi
  optional double rack_switch = 3 [default = 1.0];      // Quincy: \psi
  optional double cross_numa_node = 4 [default = 0.5];
  optional double cross_socket = 5 [default = 0.5];
  optional double cross_cache = 6 [default = 0.25];
}
